src/ngraph/serializer.cpp

```diff
@@ -851,11 +851,29 @@ static shared_ptr<ngraph::Function>
         // node->set_name(node_name);
     }
 
-    std::vector<std::shared_ptr<Node>> result;
+    //This handles both graphs w/ `op::Result` and legacy graphs w/o it
+    //If we are dealing w/ a legacy graph, add op::Result for each output node
+    ResultVector result;
+    size_t results = 0;
     for (auto result_name : func_result)
     {
-        result.push_back(node_map.at(result_name));
+        auto fr = node_map.at(result_name);
+        if (auto res = std::dynamic_pointer_cast<op::Result>(fr))
+        {
+            result.push_back(res);
+        }
+        else
+        {
+            result.push_back(std::make_shared<op::Result>(fr));
+        }
     }
+
+    if (results != 0 && results != func_result.size())
```

Comment:
Where is results getting updated?

Reply:
grrr 😢  somebody took it! fixed in the next commit.